mutex_map.go

```diff
@@ -53,6 +53,18 @@ func (m *MutexMap[K, V]) Each(callback func(key K, value V)) {
 	}
 }
 
+// Clear clears the map and runs a function for every item removed
+// the function takes in the items key and value
```

Comment:
```suggestion
// Clear removes all items from the `real` map
// Accepts an optional callback function ran for every item before it is deleted
```

@DaniElectra got the big points already, so this is all I have right now

I agree with Dani's change with making the callback optional, so this accounts for that

Also made it clear that the callback is called _before_ the call to `delete`

The callbacks parameters are also clear by the functions signature so we don't really have to repeat them here. Godoc tools, such as the Go intellisense and pkg docs, include the full signature which shows the parameters as `key` and `value`